packages/components/src/utils/is-deep-equal.ts

```diff
@@ -0,0 +1,32 @@
+import { isObject } from './jss';
+
+export const isDeepEqual = (newVal: unknown, oldVal: unknown): boolean => {
+  if (typeof newVal !== 'object' || typeof oldVal !== 'object') {
+    // string, boolean, number
+    return newVal === oldVal;
+  } else if (Array.isArray(newVal) || Array.isArray(oldVal)) {
+    return (
+      Array.isArray(newVal) &&
+      Array.isArray(oldVal) &&
+      newVal.length === oldVal.length &&
+      newVal.every((item) => oldVal.includes(item)) &&
+      oldVal.every((item) => newVal.includes(item))
```

Comment:
That's redundant with the previous one. Or how can it be that one array contains everything from another but not the other way around?